packages/stacking/src/index.ts

```diff
@@ -0,0 +1,410 @@
+   * 
+   * @returns {Promise<CoreInfo>} that resolves to a CoreInfo response if the operation succeeds
+   */
+  async getCoreInfo(): Promise<CoreInfo> {
+    const url = this.network.getInfoUrl(); 
+    return fetchPrivate(url)
+      .then((res) => res.json());
+  }
+
+  /**
+   * Get stacks node pox info
+   * 
+   * @returns {Promise<PoxInfo>} that resolves to a PoxInfo response if the operation succeeds
+   */
+  async getPoxInfo(): Promise<PoxInfo> {
+    const url = this.network.getPoxInfoUrl(); 
+    return fetchPrivate(url)
+      .then((res) => res.json());
+  }
+
+  /**
+   * Get stacks node target block time
+   * 
+   * @returns {Promise<number>} that resolves to a number if the operation succeeds
+   */
+  async getTargetBlockTime(): Promise<number>{
+    const url = this.network.getBlockTimeInfoUrl();
+    const res = await fetchPrivate(url).then((res) => res.json());
+    
+    if (this.network.isMainnet()) {
+      return res.mainnet.target_block_time;
+    } else {
+      return res.testnet.target_block_time;
+    }
+  }
+  
+  /**
+   * Get account balance
+   * 
+   * @returns {Promise<BN>} that resolves to a BigNum if the operation succeeds
+   */
+  async getAccountBalance(): Promise<BN> {
+    const url = this.network.getAccountApiUrl(this.address); 
+    return fetchPrivate(url)
+      .then((res) => res.json())
+      .then((res) => new BN(res.balance));
+  }
+
+  /**
+   * Get reward cycle duration in seconds
+   * 
+   * @returns {Promise<number>} that resolves to a number if the operation succeeds
+   */
+  async getCycleDuration(): Promise<number> {
+    const poxInfoPromise = this.getPoxInfo();
+    const targetBlockTimePromise = await this.getTargetBlockTime();
+
+    return Promise.all([poxInfoPromise, targetBlockTimePromise])
+      .then(([poxInfo, targetBlockTime]) => {
+        return poxInfo.reward_cycle_length * targetBlockTime;
+      })
+  }
+
+  /**
+   * Get number of seconds until next reward cycle
+   * 
+   * @returns {Promise<number>} that resolves to a number if the operation succeeds
+   */
+  async secondsUntilNextCycle(): Promise<number> {
+    const poxInfoPromise = this.getPoxInfo();
+    const targetBlockTimePromise = await this.getTargetBlockTime();
+    const coreInfoPromise = this.getCoreInfo();
+
+    return Promise.all([poxInfoPromise, targetBlockTimePromise, coreInfoPromise])
+      .then(([poxInfo, targetBlockTime, coreInfo]) => {
+        const blocksToNextCycle =
+        (poxInfo.reward_cycle_length -
+          ((coreInfo.burn_block_height - poxInfo.first_burnchain_block_height) %
+            poxInfo.reward_cycle_length));
+        const cycleDuration = poxInfo.reward_cycle_length * targetBlockTime;
+        return blocksToNextCycle * cycleDuration;
+      })
+  } 
+
+  /**
+   * Check if stacking is enabled for next reward cycle
+   * 
+   * @returns {Promise<boolean>} that resolves to a bool if the operation succeeds
+   */
+  async stackingEnabledNextCycle(): Promise<boolean> {
```

Comment:
would suggest `isStackingEnabledNextCycle` to be consistent. I'd not use `get`, `is` indicates the response is a boolean